packages/dev/core/src/Audio/v2/spatialAudioTransform.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable babylonjs/available */
+/* eslint-disable jsdoc/require-jsdoc */
+
+import { Quaternion, Vector3 } from "../../Maths";
+import type { TransformNode } from "../..//Meshes";
+import type { IDisposable } from "../../scene";
+import type { Nullable } from "../../types";
+
+export interface ISpatialAudioTransformOptions {
+    position?: Vector3;
+    rotation?: Quaternion;
+    attachedTransformNode?: Nullable<TransformNode>;
+}
+
+export class SpatialAudioTransform implements IDisposable {
+    private _attachedTransformNode: Nullable<TransformNode> = null;
+    private _position: Vector3;
+    private _positionDirty: boolean = false;
+    private _rotation: Quaternion;
+    private _rotationDirty: boolean = false;
+
+    public constructor(options?: ISpatialAudioTransformOptions) {
+        this._position = options?.position ?? Vector3.Zero();
+        this._rotation = options?.rotation ?? Quaternion.Identity();
+        this._attachedTransformNode = options?.attachedTransformNode ?? null;
+    }
+
+    public dispose(): void {
+        this._detachFromTransformNode();
+        this._attachedTransformNode = null;
+    }
+
+    public get position(): Vector3 {
+        this._updatePosition();
+        return this._position;
+    }
+
+    public set position(position: Vector3) {
+        this._position.copyFrom(position);
+    }
```

Comment:
This pattern is interesting. Do you expect users to be able to call `transform.position.copyFrom()`? Do you have to know when the position is dirty? Maybe the returned Vector3 should be DeepImmutable?

Reply:
I'm not sure I understand the question here. How and where would users call `transform.position.copyFrom()`?